waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -1,40 +1,41 @@
-{-# LANGUAGE ViewPatterns #-}
-
 module Wasp.Cli.Command.CreateNewProject
   ( createNewProject,
   )
 where
 
 import Control.Monad.Except (throwError)
 import Control.Monad.IO.Class (liftIO)
-import Data.Char (isLetter)
-import Data.List (intercalate)
 import StrongPath (Abs, Dir, File', Path', Rel, reldir, relfile, (</>))
 import qualified StrongPath as SP
 import System.Directory (createDirectory, getCurrentDirectory)
 import qualified System.Directory
 import qualified System.FilePath as FP
 import Text.Printf (printf)
+import Wasp.Analyzer.Parser (isValidWaspIdentifier)
 import Wasp.AppSpec.ExternalCode (SourceExternalCodeDir)
 import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Lexer (reservedNames)
 import qualified Wasp.Util.Terminal as Term
+import Data.List (intercalate)
+import Wasp.Util (indent)
 
 newtype ProjectName = ProjectName {_projectName :: String}
 
 createNewProject :: String -> Command ()
-createNewProject (all isLetter -> False) =
-  throwError $ CommandError "Project creation failed" "Please use only letters for a new project's name."
-createNewProject ((`elem` reservedNames) -> True) =
-  throwError $
-    CommandError
-      "Project creation failed"
-      ( "Please pick a project name not one of these reserved words:\n\t" ++ intercalate "\n\t" reservedNames
-      )
-createNewProject name = createNewProject' (ProjectName name)
+createNewProject name
+  | not (isValidWaspIdentifier name) =
+    throwError $
+      CommandError "Project creation failed" $
+        intercalate
+          "\n"
+          [ "The project's name must be a valid Wasp identifier:",
+            indent 2 "- It can start with a letter or an underscore.",
+            indent 2 "- It can contain only letters, numbers, or underscores.",
+            indent 2 "- It can't be a Wasp keyword: 'import', 'from', 'true', 'false'."
+          ]
+  | otherwise = createNewProject' (ProjectName name)
 
 createNewProject' :: ProjectName -> Command ()
 createNewProject' (ProjectName projectName) = do
```

Comment:
Can I express these two functions more clearly using monads? Their signature seem to fit the monadic "computations that might fail" use case.